src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -405,7 +411,12 @@ int ITupleInternal.GetHashCode(IEqualityComparer comparer)
         /// </remarks>
         public override string ToString()
         {
-            return "(" + Item1 + ")";
+            return "(" + Item1.ToString() + ")";
```

Comment:
> I thought string interpolation outperforms string concatenation

It really depends.  Let's take a few examples.

Let's say we're using short strings a, b, c, d, etc.  If you're comparing:
```$"({a})"``` and ```"(" + a + ")", then you're really comparing ```string.Format("({0})", a)``` and ```string.Concat("(", "a", ")")```.  The latter, which uses the ```string, string, string``` overload of string.Concat has very little work to do: the only allocation is the resulting string, and it only needs to get the lengths of each input string, allocate the resulting string, and copy the input strings into it.  In contrast, string.Format needs to parse the format string, call ToString on the object argument, etc.  It also needs somewhere to store the output it's building up, and for that it uses a StringBuilder.  Now, that StringBuilder is taken from the StringBuilderCache, but StringBuilderCache only supports caching StringBuilders up to a particular size, 320 chars IIRC (though it may have changed).  So, if a here is very short, the string.Format version likely won't have additional allocations beyond what the string.Concat version does, but if a ends up being long enough to go beyond the caching threshold, you could end up requiring more allocations as the StringBuilder needs to grow.

Now, let's say you're comparing ```$({a}, {b})``` and ```"(" + a + ", " + b + ")"```.  The former is the equivalent of ```string.Format("({0}, {1})", a, b)``` and the latter ```string.Concat("(", a, ", ", b, ")")```.  There is no overload of string.Concat that takes five strings, so you end up using the params string[] overload, causing this to allocate a string[] to hold the input strings.  The Concat performance though is again very quick, as it again just needs to add up the lengths of each string, allocate a new one, and copy in the inputs.  The string.Format one doesn't need that initial string array allocation, because there are only two object arguments being passed to it, not five, and thus it doesn't need to allocate a params object[], but it does still need to do all of the parsing and ToString'ing and whatnot, and it does use the StringBuilderCache and could potentially fall of the cliff and need to allocate for the StringBuilder.

This is all to say that it's a bit easier to predict the performance of the string.Concat case, especially if you don't know how big the input strings will be and whether they'll end up causing allocations for the StringBuilder.  But, string interpolation has the nice benefit of leading to very easy to read code, and in general its performance is fine.